### PR TITLE
osd/scrub: improve scheduling decision logs

### PR DESCRIPTION
**Pull request by ronen-fr**
And, to that end, relocated determine_scrub_time()
(a helper function that examines pg 'info' vs the planned
scrub details to determine the initial scrub time)
to the scrubber itself (where it belongs, and where
the logs are more expressive).

### CODE REVIEW

**Comment by ronen-fr** [comment]
jenkins run make check

**Comment by ronen-fr** [comment]
jenkins test make check

**Comment by ronen-fr** [comment]
jenkins test windows

**Comment by ronen-fr** [comment]
jenkins test windows

File: src/osd/scrubber/pg_scrubber.h [hunk 1]
```diff
@@ -784,6 +784,15 @@ class PgScrubber : public ScrubPgIF,
    */
   Scrub::sched_conf_t populate_config_params() const;
 
+  /**
+   * determine the time when the next scrub should be scheduled
+   *
+   * based on the planned scrub's flags, time of last scrub, and
+   * the pool's scrub configuration.
+   */
+  Scrub::sched_params_t determine_scrub_time(
```
Comment (rzarzynski) [inline src/osd/scrubber/pg_scrubber.h:793]: (assuming the first commit is just code shuffling)

File: src/osd/scrubber/pg_scrubber.cc [hunk 2]
```diff
@@ -606,11 +606,12 @@ scrub_level_t PgScrubber::scrub_requested(
   const bool deep_requested = (scrub_level == scrub_level_t::deep) ||
 			      (scrub_type == scrub_type_t::do_repair);
   dout(10) << fmt::format(
-		  "{}: {} {} scrub requested. Prev stamp: {}. Registered? {}",
+		  "{}: {}{} scrub requested. "
+		  "@entry:{},last-stamp:{:s}.,Registered?{}",
```
Comment (rzarzynski) [inline src/osd/scrubber/pg_scrubber.cc:610]: ACK.

File: src/osd/scrubber/pg_scrubber.cc [hunk 3]
```diff
@@ -606,11 +606,12 @@ scrub_level_t PgScrubber::scrub_requested(
...
 		  __func__,
-		  (scrub_type == scrub_type_t::do_repair ? " repair + "
-							 : " not-repair + "),
-		  (deep_requested ? "deep" : "shallow"),
+		  (scrub_type == scrub_type_t::do_repair ? "repair + "
+							 : "not-repair + "),
+		  (deep_requested ? "deep" : "shallow"), req_flags,
```
Comment (rzarzynski) [inline src/osd/scrubber/pg_scrubber.cc:614]: `req_flags` is easy to miss. Maybe worth a new line?

**Review by rzarzynski — APPROVED** [review]

File: src/osd/scrubber/pg_scrubber.cc (hunk 3 above)
Comment (ronen-fr) [reply src/osd/scrubber/pg_scrubber.cc:614]: A separate line in the code, I assume (not the log?)
The 'req_flags' will be removed shortly (and, besides, that's how clang-format set the line)

File: src/osd/scrubber/pg_scrubber.cc (hunk 2 above)
Comment (ronen-fr) [reply src/osd/scrubber/pg_scrubber.cc:610]: Helped me noticed the extra '.'...

**Comment by ronen-fr** [comment]
jenkins test make check

**Comment by ronen-fr** [comment]
Merging based on my Teuthology tests

**Pull request merged**